inlong-sort/sort-core/src/main/java/org/apache/inlong/sort/parser/impl/FlinkSqlParser.java

```diff
@@ -890,10 +910,13 @@ private String parseFields(List<FieldInfo> fields, Node node) {
      * Generate primary key format in sql
      *
      * @param primaryKey The primary key of table
+     * @param filterPrimaryKey filter PrimaryKey, use for mongo
      * @return Primary key format in sql
      */
-    private String genPrimaryKey(String primaryKey) {
-        if (StringUtils.isNotBlank(primaryKey)) {
+    private String genPrimaryKey(String primaryKey, String filterPrimaryKey) {
+        boolean checkPrimaryKeyFlag = StringUtils.isNotBlank(primaryKey)
+                && (StringUtils.isBlank(filterPrimaryKey) || !primaryKey.contains(filterPrimaryKey));
```

Comment:
Will there is problem using the `contains` method? For example, when primaryKey="global_id" and filterPrimaryKey="id".

Reply:
@kuansix you need modify to use `equals`